packages/bcer-api/app/src/notification/textService.ts

```diff
@@ -0,0 +1,77 @@
+import { NotAcceptableException } from '@nestjs/common';
+import { NotifyClient } from 'notifications-node-client';
+import { GeneralUtil, sleep } from 'src/utils/util';
+import { NotificationReportDTO } from './dto/notification-report.dto';
+import { NotificationDTO } from './dto/notification.dto';
+
+export class TextService {
+  private readonly apiEndpoint =
+    process.env.TEXT_API_ENDPOINT || 'https://api.notification.canada.ca';
+  private readonly apiKey = process.env.TEXT_API_KEY;
+  private readonly templateId =
+    process.env.TEXT_GENERIC_NOTIFICATION_TEMPLATE_ID;
+  private readonly reference = process.env.TEXT_REFERENCE || 'abc123';
+
+  private messageClient: any;
+
+  constructor() {
+    this.messageClient = new NotifyClient(this.apiEndpoint, this.apiKey);
+  }
+
+  private async send(message: string, phoneNumber: string) {
+    const response = await this.messageClient.sendSms(
+      this.templateId,
+      `${phoneNumber}`,
+      {
+        personalisation: {
+          message,
+        },
+        reference: this.reference,
+      },
+    );
+  }
+
+  async sendMessage(
+    data: NotificationDTO,
+    phoneNumbers: Array<string>,
+  ): Promise<NotificationReportDTO> {
+    const { message, title } = data;
+    if (message?.length > 612) throw NotAcceptableException;
+    const result = await (Promise as any).allSettled(
+      phoneNumbers.map(async (phoneNumber, i) => {
+        /**
+         * Limiting the api request to a maximum of 1000 per second.
+         */
+        await sleep(Math.floor(i + 1) / 1000);
```

Comment:
Looks like the API limit is actually `1000 per minute` 😓  So we might have to rethink this one a bit!

https://documentation.notification.canada.ca/en/limits.html#daily-limits

Reply:
😢